src/plugins/controls/public/control_group/embeddable/control_group_container.tsx

```diff
@@ -127,68 +119,11 @@ export class ControlGroupContainer extends Container<
     return this.createAndSaveEmbeddable(panelState.type, panelState);
   }
 
-  /**
-   * Returns a button that allows controls to be created externally using the embeddable
-   * @param buttonType Controls the button styling
-   * @param closePopover Closes the create control menu popover when flyout opens - only necessary if `buttonType === 'toolbar'`
-   * @return If `buttonType == 'toolbar'`, returns `EuiContextMenuPanel` with input control types as items.
-   *         Otherwise, if `buttonType == 'callout'` returns `EuiButton` with popover containing input control types.
-   */
-  public getCreateControlButton = (
-    buttonType: CreateControlButtonTypes,
-    closePopover?: () => void
-  ) => {
-    const ControlsServicesProvider = pluginServices.getContextProvider();
-
-    return (
-      <ControlsServicesProvider>
-        <CreateControlButton
-          buttonType={buttonType}
-          defaultControlWidth={this.getInput().defaultControlWidth}
-          defaultControlGrow={this.getInput().defaultControlGrow}
-          updateDefaultWidth={(defaultControlWidth) => this.updateInput({ defaultControlWidth })}
-          updateDefaultGrow={(defaultControlGrow: boolean) =>
-            this.updateInput({ defaultControlGrow })
-          }
-          addNewEmbeddable={(type, input) => {
-            if (type === OPTIONS_LIST_CONTROL) {
-              this.addOptionsListControl(input as AddOptionsListControlProps);
-              return;
-            }
-
-            if (type === RANGE_SLIDER_CONTROL) {
-              this.addRangeSliderControl(input as AddRangeSliderControlProps);
-              return;
-            }
-
-            this.addDataControlFromField(input as AddDataControlProps);
-          }}
-          closePopover={closePopover}
-          getRelevantDataViewId={() => this.getMostRelevantDataViewId()}
-          setLastUsedDataViewId={(newId) => this.setLastUsedDataViewId(newId)}
-        />
-      </ControlsServicesProvider>
-    );
-  };
-
-  public getCreateTimeSliderControlButton = (closePopover?: () => void) => {
-    const childIds = this.getChildIds();
-    const hasTimeSliderControl = childIds.some((id) => {
-      const child = this.getChild(id);
-      return child.type === TIME_SLIDER_CONTROL;
-    });
-    return (
-      <CreateTimeSliderControlButton
-        onCreate={() => {
-          this.addTimeSliderControl();
-        }}
-        closePopover={closePopover}
-        hasTimeSliderControl={hasTimeSliderControl}
-      />
-    );
-  };
+  public openAddDataControlFlyout() {
+    openAddDataControlFlyout(this);
+  }
 
-  private getEditControlGroupButton = (closePopover: () => void) => {
+  public getEditControlGroupButton = (closePopover: () => void) => {
```

Comment:
I wonder if we should standardize this to match the `openAddDataControlFlyout` in a follow up. It could also be a method call named something like `showControlGroupSettings` which opens the editor flyout with all context. Then consumers would be in charge of the styling of the button.